src/new-contract-interaction/new-contract-interaction.js

```diff
@@ -0,0 +1,163 @@
+      return findings;
+    }
+
+    // get all addresses involved with this transaction that are non-Aave contracts
+    let addresses = Object.keys(txEvent.addresses);
+
+    // to minimize Etherscan requests (slow and rate limited to 5/sec for the free tier),
+    // exclude the lending pool address, the incentives controller, and all token addresses
+    let exclusions = [
+      lendingPoolV2Address,
+      contractAddresses.IncentivesController.toLowerCase(),
+    ];
+    exclusions = exclusions.concat(tokenAddresses);
+    addresses = addresses.filter((item) => exclusions.indexOf(item) < 0);
+
+    // watch for recently created contracts interacting with Aave lending pool
+    if (txEvent.transaction.to === lendingPoolV2Address) {
+      // create an array of promises that retrive the contract code for each address
+      const contractCodePromises = [];
+      let contractCode = [];
+      addresses.forEach((address) => {
+        // RPC call for contract code
+        const codePromise = ethersProvider.getCode(address);
+
+        // associate each address with the code that was returned
+        codePromise.then((result) => contractCode.push({ address, code: result }));
+
+        // to prevent Promise.all() from rejecting, catch failed promises and set the return
+        // value to undefined
+        contractCodePromises.push(codePromise.catch(() => undefined));
+      });
+
+      // resolve the requests
+      await Promise.all(contractCodePromises);
+
+      // filter out EOAs from our original list of addresses
+      contractCode = contractCode.filter((item) => (item.code !== '0x'));
+      addresses = contractCode.map((item) => item.address);
+
+      // Next, for each contract, query the Etherscan API for the list of transactions
+      // the first transaction item returned in the results will be the earliest
+      const etherscanTxlistPromises = [];
+      const txData = [];
+      contractCode.forEach((item) => {
+        const txlistPromise = axios.get(baseUrl + item.address + options + apiKey);
+
+        // associate each address with the transaction list that was returned
+        txlistPromise.then((result) => txData.push({ address: item.address, response: result }));
+
+        // to prevent Promise.all() from rejecting, catch failed promises and set the return
+        // value to undefined
+        etherscanTxlistPromises.push(txlistPromise.catch(() => undefined));
+      });
+
+      // resolve the requests
+      await Promise.all(etherscanTxlistPromises);
+
+      // process the results
+      txData.forEach((item) => {
+        // bail if the API did not return a valid result
+        if (item.response.data.status === 0) {
+          return;
+        }
+
+        // get the timestamp from the earliest transaction
+        const creationTime = item.response.data.result[0].timeStamp;
+
+        // compute days elapsed since contract creation
+        const currentTime = Date.now() / 1000;
```

Comment:
I'm not certain that we want to rely on the timestamp of the Scan Agent.  Perhaps:
```suggestion
        const currentTime = txEvent.timestamp;
```

Reply:
Changed